ios/browser/api/password/brave_password_api.h

```diff
@@ -0,0 +1,123 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_PASSWORD_BRAVE_PASSWORD_API_H_
+#define BRAVE_IOS_BROWSER_API_PASSWORD_BRAVE_PASSWORD_API_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, PasswordFormScheme) {
+  PasswordFormSchemeTypeHtml = 0,
+  PasswordFormSchemeTypeBasic,
+  PasswordFormSchemeTypeDigest,
+  PasswordFormSchemeTypeOther,
+  PasswordFormSchemeMobile,
+  PasswordFormSchemeUsernameOnly
+};
+
+@protocol PasswordStoreObserver;
+@protocol PasswordStoreListener;
+
+@class IOSPasswordForm;
+
+NS_SWIFT_NAME(PasswordForm)
+OBJC_EXPORT
+@interface IOSPasswordForm : NSObject <NSCopying>
+
+@property(nonatomic, strong) NSURL* url;
+@property(nonatomic, strong) NSString* signOnRealm;
+@property(nonatomic, nullable, copy) NSDate* dateCreated;
+@property(nonatomic, nullable, copy) NSDate* dateLastUsed;
+@property(nonatomic, nullable, copy) NSDate* datePasswordChanged;
+@property(nonatomic, nullable, copy) NSString* usernameElement;
+@property(nonatomic, nullable, copy) NSString* usernameValue;
+@property(nonatomic, nullable, copy) NSString* passwordElement;
+@property(nonatomic, nullable, copy) NSString* passwordValue;
+@property(nonatomic) bool isBlockedByUser;
+@property(nonatomic) PasswordFormScheme scheme;
+
+/// Password Form Constructor used with BravePasswordAPI
+/// @param url - Primary data used by the PasswordManager to decide (in longest
+///        matching prefix fashion) whether or not a given PasswordForm
```

Comment:
not sure where this is handled, but it is very important security-wise that iOS never autofills a password unless the URL realm (scheme + host + port) of the password database entry is the same as the current origin of the document that the form is loaded in. (not necessarily the top level page URL, because the form may be in a cross-origin iframe)

it is definitely worth adding tests for this

Reply:
This is handled entirely on iOS side and LoginsHelper script from FF is injected for `AutoFill`. 

We made some improvements to solve iFrame and Cross-Site-Scripting problem https://github.com/brave/brave-ios/issues/4113 while we were adding screens for local logins and passwords https://github.com/brave/brave-ios/pull/4124.

In this PR we are only bridging the password manager to iOS so we can use entirely Chromium on iOS side. I will check the unit tests for LoginsHelper on iOS side and add the necessary tests there when the iOS side PR is ready. And I will open another security review for that PR. (In draft state now https://github.com/brave/brave-ios/pull/5013)